Refactoring of all functional tests under `functest` dir

aruniiird opened the pull request:
Signed-off-by: Arun Kumar Mohan <amohan@redhat.com>

aruniiird commented:
@umangachapagain , @jarrpa 
Need an `ok-to-test` kick start the CI checks

umangachapagain commented:
/ok-to-test

aruniiird commented:
@jarrpa , I have divided the commits into several logical ones. Please take a look...

jarrpa commented on functests/storageclusterinitialization_test.go:
```diff
@@ -483,6 +512,44 @@ func (scExpctObj *storageClassExpect) ExpectReconcile() error {
+		scInitObj.createCephFilesystemExpect(tobeDeleted),
+		scInitObj.createStorageClassExpect(tobeDeleted),
+	}
+	// CephBlockPool can only be added if it is not tobe deleted (ie; 'tobeDeleted' flag is false)
```
```suggestion
	// CephBlockPool can only be added if it is not to be deleted (ie; 'tobeDeleted' flag is false)
```

jarrpa commented on functests/storageclusterinitialization_test.go:
```diff
@@ -525,123 +592,43 @@ func StorageClusterInitializationTest() {
-					err = cephObjStoreUserExpect.ExpectModify()
+				var expectResources = getAllValidExpectResources(scInitObj, true)
+				for _, eachResource := range expectResources {
+					By(fmt.Sprintf("Modifying %s (expected tobe deleted)",
```
```suggestion
					By(fmt.Sprintf("Modifying %s (expected to be deleted)",
```

aruniiird commented:
I have addressed all the review comments except this one,

> For your last two commits, is it possible to simply have one struct Expect{} type that takes a kind string and an obj runtime.Object?

I am unable to generalize the functions `ExpectReconcile()` and `ExpectModify()` , with the `runtime.Object` interface, as these functions require specific / concrete object's `Spec` values (like `Gateway` or `DataPools[]` or `Store` etc). Example lines,

https://github.com/openshift/ocs-operator/blob/95db65455429b7c93f399cc3e41f4d6c29d9d2b7/functests/storageclusterinitialization_test.go#L361
https://github.com/openshift/ocs-operator/blob/95db65455429b7c93f399cc3e41f4d6c29d9d2b7/functests/storageclusterinitialization_test.go#L287
https://github.com/openshift/ocs-operator/blob/95db65455429b7c93f399cc3e41f4d6c29d9d2b7/functests/storageclusterinitialization_test.go#L213

Solutions which I come up with to generalize all objects (bringing another common interfaces to get the specific `Spec`'s value) are making the tests more and more complicated.
